packages/waku/src/lib/hono/runner.ts

```diff
@@ -53,10 +53,12 @@ export const runner = (options: MiddlewareOptions): MiddlewareHandler => {
       });
     };
     await run(0);
-    return c.body(
-      ctx.res.body || null,
-      (ctx.res.status as any) || 200,
-      ctx.res.headers || {},
-    );
+    if (!c.finalized || ctx.res.body) {
```

Comment:
@dai-shi Checking for `!c.finalized || ctx.res.body` instead of just `!c.finalized` gets the e2e tests to pass again.

Reply:
Thanks for investigating!
Hmm, but is there anything weird in the e2e tests?
I hope to avoid checking `ctx`, but only `c`... 🤔

Reply:
I see. It looks like the dev server in the e2e tests has an issue. I am seeing it return a 404 error instead of the page response when I use the playwright vscode extension to debug rsc-basic.spec.ts.

I'll comment out the workaround and let the e2e tests fail until the root issue can be found with the e2e tests.

Reply:
@himself65 do you have any idea?